lib/mount_util.c

```diff
@@ -65,9 +64,6 @@ static int mtab_needs_update(const char *mnt)
 		uid_t ruid;
 		int err;
 
-		if (S_ISLNK(stbuf.st_mode))
-			return 0;
-
```

Comment:
Ok, I see this  is what the symlink discussion is about - we first need to check if there would be side effects.

Reply:
I wonder if any of these mtab checks are actually necessary, I would assume that libmount would not try to edit the tab if its nonexistent/unwritable?

Reply:
I really don't know. So this line comes from commit 5f28cd15ab43c741f6d116be4d3a9aa5d82ab385, which explains what the code does, but not why...

```
commit 5f28cd15ab43c741f6d116be4d3a9aa5d82ab385
Author: Miklos Szeredi <miklos@szeredi.hu>
Date:   Thu Jul 10 19:35:21 2008 +0000

    Skip calling mount(8) if /etc/mtab doesn't exist...
```

Exactly the reason why I typically insist on more verbose commit messages. Well, the commit also has a ChangeLog entry

```
diff --git a/ChangeLog b/ChangeLog
index 638dd3cb..4f565668 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -3,6 +3,10 @@
        * Fix handling of (no)suid and (no)dev options if filesystem is
        mounted from /etc/fstab or via mount(8).  Reported by Jan Ondrej.
 
+       * Skip calling mount(8) if /etc/mtab doesn't exist or if it's on a
+       read-only filesystem.  This works around issues with certain mount
+       implementations.  Reported by Szabolcs Szakacsits.
+
```

No idea if these issues do still exist - I think any recent distribution has mtab as symlink...

Reply:
I actually start to wonder if libfuse should link to libmount, because there are functions available that let you check for a writable utab. But certainly not in this commit, just something we might consider in the future.